app/assets/javascripts/actions/tickets_actions.js

```diff
@@ -149,6 +151,32 @@ export const deleteTicket = id => async (dispatch) => {
   dispatch({ type: DELETE_TICKET, id });
 };
 
+
+export const deleteNotePromise = (id) => {
+  return new Promise((res, rej) => {
+    return $.ajax({
+      type: 'DELETE',
+      url: `/td/tickets/replies/${id}`,
+      success: function (data) {
+        return res(data);
+      }
+    })
+    .fail((err) => {
+      logErrorMessage(err);
+      return rej(err);
+    });
+  });
+};
+
+export const deleteNote = id => (dispatch) => {
+    deleteNotePromise(id)
+      .then(() => {
+        dispatch({ type: MESSAGE_KIND_NOTE_DELETE, id });
+        window.location.reload();
```

Comment:
Is reloading the window necessary? Updating the store via the MESSAGE_KIND_NOTE_DELETE action ought to change the state and trigger a refresh so that the note disappears.

Reply:
Its not necessary, but I thought it will be better experience, that reloading a page once note has been deleted. I feel it won't be that much good user experience if we show **success alert** compared to reloading of page, once item gets deleted. MESSAGE_KIND_NOTE_DELETE action is only called if the **delete**  request is successful, then once the state changes in redux store, then I refresh the page. What do you suggest for this case ?

Reply:
I think a success alert is a better experience than a page reload, as a page reload might take several seconds (depending on connection speed), and a big part of using React is that we shouldn't need to reload the whole page very often.

Reply:
True, I'll do that